### PR TITLE
[Docs] Refactor the docs

### PR DESCRIPTION
**Pull request by vkorukanti**
Refactor the docs to:
* make it easy to access the documentation for each connector.
* add Kernel docs

Changes are staged at: https://docs.delta.io/0.0.2/index.html

### CODE REVIEW

File: docs/source/best-practices.md [hunk 1]
```diff
@@ -186,4 +186,5 @@ You should not use [Spark caching](optimizations/delta-cache.md#delta-and-rdd-ca
 
 - The data that gets cached may not be updated if the table is accessed using a different identifier (for example, you do `spark.table(x).cache()` but then write to the table using `spark.write.save(/some/path)`.
 
-.. include:: /shared/replacements.md
+.. <Delta> replace:: Delta Lake
```
Comment (tdas) [inline docs/source/best-practices.md:189]: why replace this?

Comment (tdas) [reply docs/source/best-practices.md:189]: seems like a lot of duplication now.

File: docs/source/conf.py [hunk 2]
```diff
@@ -110,7 +110,15 @@
 html_logo = "_static/delta-lake-white.png"
 html_favicon = "_static/favicon.ico"
 
-html_theme_options = {}
+html_theme_options = {
+    "show_toc_level": 2,
```
Comment (tdas) [inline docs/source/conf.py:114]: what is this for? isnt this supposed to show 2 levels always expanded? is it not working?

File: docs/source/conf.py [hunk 3]
```diff
@@ -110,7 +110,15 @@
...
+    "use_sidenav_for_toctree": True,
```
Comment (tdas) [inline docs/source/conf.py:115]: can comment what each of these setting are for future devs.

File: docs/source/best-practices.md (hunk 1 above)
Comment (vkorukanti) [reply docs/source/best-practices.md:189]: Don't know the reason, but having the `./shared/replacement.md` in every file cause the side navigation to break.

File: docs/source/conf.py (hunk 2 above)
Comment (vkorukanti) [reply docs/source/conf.py]: removed this block.

File: docs/source/conf.py (hunk 3 above)
Comment (vkorukanti) [reply docs/source/conf.py]: remove this entire block. its not needed.

File: docs/source/conf.py [hunk 4]
```diff
@@ -110,7 +110,15 @@
...
+    "navigation_depth": 2
+}
+
+html_sidebars = {
```
Comment (vkorukanti) [inline docs/source/conf.py]: remove this as well. not needed.

File: docs/source/best-practices.md (hunk 1 above)
Comment (tdas) [reply docs/source/best-practices.md:189]: WOW.

**Review by tdas — APPROVED** [review]

**Pull request merged**